presto-main/src/main/java/com/facebook/presto/execution/SqlTask.java

```diff
@@ -373,6 +375,11 @@ private TaskInfo createTaskInfo(TaskHolder taskHolder)
         return Futures.transform(futureTaskState, input -> getTaskInfo(), directExecutor());
     }
 
+    public void updateCommunicationSlug(String communicationSlug)
```

Comment:
Should be set instead of update? We should only set it once but can update many times.